ci: failure in `wallet_importdescriptors.py `

https://github.com/bitcoin/bitcoin/actions/runs/27626623050/job/81689877009?pr=35537:
```bash
2026-06-16T15:17:31.3998250Z [0;34m node0 2026-06-16T15:17:21.699894Z [http.00] [wallet/sqlite.cpp:55] [TraceSqlCallback] [walletdb:trace] [/Users/runner/work/bitcoin/bitcoin/repo_archive/ci/scratch_ ₿🧪_/test_runner/test_runner_₿_🏃_20260616_150224/wallet_importdescriptors_103/node0/regtest/wallets/abort_import_wallet/wallet.dat] SQLite Statement: INSERT or REPLACE into main values(?, ?) [0m
2026-06-16T15:17:31.3999070Z [0;34m node0 2026-06-16T15:17:21.699970Z [http.00] [wallet/wallet.h:939] [WalletLogPrintf] [abort_import_wallet] RescanFromTime: Rescanning last 329 blocks [0m
2026-06-16T15:17:31.3999900Z [0;34m node0 2026-06-16T15:17:21.699981Z [http.00] [wallet/wallet.h:939] [WalletLogPrintf] [abort_import_wallet] Rescan started from block 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206... (slow variant inspecting all blocks) [0m
2026-06-16T15:17:31.4000360Z [0;34m node0 2026-06-16T15:17:21.705100Z [http.00] [wallet/wallet.h:939] [WalletLogPrintf] [abort_import_wallet] Scanning current mempool transactions. [0m
2026-06-16T15:17:31.4000970Z [0;34m node0 2026-06-16T15:17:21.705121Z [http.00] [wallet/wallet.h:939] [WalletLogPrintf] [abort_import_wallet] Rescan completed in               5ms [0m
2026-06-16T15:17:31.4001230Z [0;36m test  2026-06-16T15:17:21.705640Z TestFramework (ERROR): Unexpected exception: [0m
2026-06-16T15:17:31.4001400Z [0;36m                                   Traceback (most recent call last):[0m
2026-06-16T15:17:31.4001830Z [0;36m                                     File "/Users/runner/work/bitcoin/bitcoin/repo_archive/test/functional/test_framework/test_framework.py", line 144, in main[0m
2026-06-16T15:17:31.4001970Z [0;36m                                       self.run_test()[0m
2026-06-16T15:17:31.4002090Z [0;36m                                       ~~~~~~~~~~~~~^^[0m
2026-06-16T15:17:31.4002700Z [0;36m                                     File "/Users/runner/work/bitcoin/bitcoin/repo_archive/ci/scratch_ ₿🧪_/build-aarch64-apple-darwin24.6.0/test/functional/wallet_importdescriptors.py", line 962, in run_test[0m
2026-06-16T15:17:31.4002880Z [0;36m                                       self.test_rescan_fails_import()[0m
2026-06-16T15:17:31.4003060Z [0;36m                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^[0m
2026-06-16T15:17:31.4003710Z [0;36m                                     File "/Users/runner/work/bitcoin/bitcoin/repo_archive/ci/scratch_ ₿🧪_/build-aarch64-apple-darwin24.6.0/test/functional/wallet_importdescriptors.py", line 193, in test_rescan_fails_import[0m
2026-06-16T15:17:31.4003990Z [0;36m                                       raise AssertionError("importdescriptors unexpectedly succeeded")[0m
2026-06-16T15:17:31.4004520Z [0;36m                                   AssertionError: importdescriptors unexpectedly succeeded[0m
2026-06-16T15:17:31.4004800Z [0;36m test  2026-06-16T15:17:21.709227Z TestFramework (DEBUG): Closing down network thread [0m
```

```
2026-06-16T15:17:22.3162210Z 2026-06-16T15:17:21.137717Z TestFramework (INFO): Aborting an importdescriptors rescan should fail the RPC call
2026-06-16T15:17:22.3162690Z 2026-06-16T15:17:21.705640Z TestFramework (ERROR): Unexpected exception:
2026-06-16T15:17:22.3163000Z Traceback (most recent call last):
2026-06-16T15:17:22.3163450Z   File "/Users/runner/work/bitcoin/bitcoin/repo_archive/test/functional/test_framework/test_framework.py", line 144, in main
2026-06-16T15:17:22.3163980Z     self.run_test()
2026-06-16T15:17:22.3164120Z     ~~~~~~~~~~~~~^^
2026-06-16T15:17:22.3164940Z   File "/Users/runner/work/bitcoin/bitcoin/repo_archive/ci/scratch_ ₿🧪_/build-aarch64-apple-darwin24.6.0/test/functional/wallet_importdescriptors.py", line 962, in run_test
2026-06-16T15:17:22.3165530Z     self.test_rescan_fails_import()
2026-06-16T15:17:22.3165780Z     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
2026-06-16T15:17:22.3171280Z   File "/Users/runner/work/bitcoin/bitcoin/repo_archive/ci/scratch_ ₿🧪_/build-aarch64-apple-darwin24.6.0/test/functional/wallet_importdescriptors.py", line 193, in test_rescan_fails_import
2026-06-16T15:17:22.3173720Z     raise AssertionError("importdescriptors unexpectedly succeeded")
2026-06-16T15:17:22.3175000Z AssertionError: importdescriptors unexpectedly succeeded
```
The test failing is the "abort rescan" case.
The test fails due to a race condition in the c++ code. It can happen that an `abortrescan` call is ignored by the actual rescan code.

 #35512 should fix it by moving the code that makes the rescan ignore the `abortrescan` call

Another failure, same test but not the race patched in #35512 

https://github.com/bitcoin/bitcoin/actions/runs/27913805625/job/82595425742?pr=35182

Failed in the ancestor commits test. I don't think its related to the PR because at that commit nothing significant has been changed in the HTTP server (just implementing new classes)

```
2026-06-21T19:27:55.6660429Z  test  2026-06-21T19:27:49.552619Z TestFramework (INFO): Test importdescriptors fails when wallet is already rescanning 
...
2026-06-21T19:27:57.2336099Z  node0 2026-06-21T19:27:52.326674Z [http.01] [wallet/wallet.h:939] [WalletLogPrintf] [rescan_wallet] Scanning current mempool transactions. 
2026-06-21T19:27:57.2336326Z  node0 2026-06-21T19:27:52.326680Z [http.01] [wallet/wallet.h:939] [WalletLogPrintf] [rescan_wallet] Rescan completed in              35ms 
2026-06-21T19:27:57.2336466Z  test  2026-06-21T19:27:52.327147Z TestFramework (ERROR): Unexpected exception: 
2026-06-21T19:27:57.2336557Z                                    Traceback (most recent call last):
2026-06-21T19:27:57.2336897Z                                      File "/home/runner/work/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 145, in main
2026-06-21T19:27:57.2336971Z                                        self.run_test()
2026-06-21T19:27:57.2337208Z                                      File "/home/runner/work/bitcoin/bitcoin/ci_build/test/functional/wallet_importdescriptors.py", line 962, in run_test
2026-06-21T19:27:57.2337298Z                                        self.test_rescan_fails_import()
2026-06-21T19:27:57.2337565Z                                      File "/home/runner/work/bitcoin/bitcoin/ci_build/test/functional/wallet_importdescriptors.py", line 164, in test_rescan_fails_import
2026-06-21T19:27:57.2337648Z                                        assert_equal(num_success, 1)
2026-06-21T19:27:57.2337883Z                                      File "/home/runner/work/bitcoin/bitcoin/test/functional/test_framework/util.py", line 94, in assert_equal
2026-06-21T19:27:57.2338158Z                                        raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
2026-06-21T19:27:57.2338282Z                                    AssertionError: not(2 == 1)
```

> Another failure, same test but not the race patched in [#35512](https://github.com/bitcoin/bitcoin/pull/35512)

Happened on [FreeBSD 14.4](https://github.com/hebasto/bitcoin-core-nightly/actions/runs/28314030917/job/83883847073) as well:
```
2026-06-28T07:54:41.768861Z TestFramework (INFO): Test older() safety
2026-06-28T07:54:41.797294Z TestFramework (INFO): Test import of unused(KEY)
2026-06-28T07:54:41.988492Z TestFramework (INFO): Test import of unused(KEY) with existing KEY
2026-06-28T07:54:42.007588Z TestFramework (INFO): Test import of unused(KEY) to wallet without privkeys
2026-06-28T07:54:42.028605Z TestFramework (INFO): Test importdescriptors fails when wallet is already rescanning
2026-06-28T07:54:47.258072Z TestFramework (ERROR): Unexpected exception:
Traceback (most recent call last):
  File "/root/bitcoin/test/functional/test_framework/test_framework.py", line 145, in main
    self.run_test()
  File "/root/build/test/functional/wallet_importdescriptors.py", line 994, in run_test
    self.test_rescan_fails_import()
  File "/root/build/test/functional/wallet_importdescriptors.py", line 168, in test_rescan_fails_import
    assert_equal(num_success, 1)
  File "/root/bitcoin/test/functional/test_framework/util.py", line 94, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(2 == 1)
```

... and on [NetBSD 10.1](https://github.com/hebasto/bitcoin-core-nightly/actions/runs/28770350665/job/85302709800):
```
2026-07-06T07:09:32.900122Z TestFramework (INFO): Test older() safety
2026-07-06T07:09:33.011553Z TestFramework (INFO): Test import of unused(KEY)
2026-07-06T07:09:33.918427Z TestFramework (INFO): Test import of unused(KEY) with existing KEY
2026-07-06T07:09:35.198930Z TestFramework (INFO): Test import of unused(KEY) to wallet without privkeys
2026-07-06T07:09:35.462252Z TestFramework (INFO): Test importdescriptors fails when wallet is already rescanning
2026-07-06T07:10:26.660729Z TestFramework (ERROR): Unexpected exception:
Traceback (most recent call last):
  File "/root/bitcoin/test/functional/test_framework/test_framework.py", line 145, in main
    self.run_test()
    ~~~~~~~~~~~~~^^
  File "/root/build/test/functional/wallet_importdescriptors.py", line 994, in run_test
    self.test_rescan_fails_import()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/root/build/test/functional/wallet_importdescriptors.py", line 168, in test_rescan_fails_import
    assert_equal(num_success, 1)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/root/bitcoin/test/functional/test_framework/util.py", line 94, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(2 == 1)
```

--- a/src/wallet/rpc/backup.cpp
+++ b/src/wallet/rpc/backup.cpp
@@ -380,14 +380,14 @@ RPCMethod importdescriptors()
     if (!pwallet) return UniValue::VNULL;
     CWallet& wallet{*pwallet};
 
-    // Make sure the results are valid at least up to the most recent block
-    // the user could have gotten from another RPC command prior to now
-    wallet.BlockUntilSyncedToCurrentChain();
-
     WalletRescanReserver reserver(*pwallet);
     if (!reserver.reserve(/*with_passphrase=*/true)) {
         throw JSONRPCError(RPC_WALLET_ERROR, "Wallet is currently rescanning. Abort existing rescan or wait.");
     }
+
+    // Make sure the results are valid at least up to the most recent block
+    // the user could have gotten from another RPC command prior to now
+    wallet.BlockUntilSyncedToCurrentChain();
 
     // Ensure that the wallet is not locked for the remainder of this RPC, as
     // the passphrase is used to top up the keypool.

--- a/test/functional/wallet_importdescriptors.py
+++ b/test/functional/wallet_importdescriptors.py
@@ -136,6 +136,9 @@ class ImportDescriptorsTest(BitcoinTestFramework):
             "timestamp": 0, "range": [0, 10000]}]
             conflicting_desc = [{"desc": descsum_create("pkh(" + xpriv + "/1h/*h)"),
             "timestamp": 0, "range": [0, 10000]}]
+            num_relevant_blocks = 1000
+            self.generatetoaddress(self.nodes[0], num_relevant_blocks, self.nodes[0].deriveaddresses(slow_desc[0]['desc'], [0, 0])[0])
+            self.generatetoaddress(self.nodes[0], num_relevant_blocks, self.nodes[0].deriveaddresses(conflicting_desc[0]['desc'], [0, 0])[0])
 
             start = threading.Barrier(3)
 

